cluster/images/conformance/run_e2e.sh

```diff
@@ -13,7 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -o errexit
```

Comment:
it'd be better to change the `wait` line to be something like
```bash
wait "$(pgrep ginkgo)" && ret=0 || ret=$?
```
and then the last line of this script would be 
```bash
exit ${ret}
```

Reply:
To clarify to ensure I understand correctly,

is the reason you want to do that so that it isn't just customized for Sonobuoy (waiting for the done file) but also a consumer could wait for the non-zero exit code for indication of pass/fail?

I guess those two things are at odds: 
 - if we set errexit and the script fails anywhere results wont be "published" and Sonobuoy would hang
 - if we dont set errexit and the script fails anywhere the exit code wont be properly reported and it might report success

If my understanding above is accurate, I agree that the lesser evil is to do as you suggest. A hang would at least be indicative of a problem whereas a false-positive is a larger issue that could go undiscovered.

Also +1/TIL to that particular bash logic; I dont think I'd seen that _exact_ trick used before but it seems pretty nifty.

Reply:
we generally like keeping `errexit` on scripts so that we fail fast on syntax errors or other issues - you're sorta required to handle all error cases, or everything will just fail.

another alternative if we want to make sure logs are always uploaded is to add something like
```
trap saveResults EXIT
```

and then you wouldn't even need to explicitly call `saveResults` - it'd always run on script exit.